src/components/OtpInput.js

```diff
@@ -0,0 +1,213 @@
+    }),
+    [isEmpty, validateData]
+  );
+
+  const [otpInputId] = useState(() => `otp-input-${nanoid()}`);
+  const [auxId] = useState(() => `otp-input-aux-${nanoid()}`);
+
+  const {
+    value,
+    errors,
+    hasErrors,
+    onFocus,
+    onBlur,
+    onChange: fieldOnChange,
+  } = useField("OtpInput", {
+    name,
+    disabled,
+    optional,
+    validate,
+    data: validationData,
+  });
+
+  const onChange = useCallback(
+    (otpValue) => {
+      fieldOnChange({ target: { value: otpValue, name } });
+      onChangeProp && onChangeProp(otpValue);
+    },
+    [fieldOnChange, onChangeProp, name]
+  );
+
+  return (
+    <Field
+      optional={optional}
+      disabled={disabled}
+      label={label}
+      labelFor={otpInputId}
+      auxId={auxId}
+      helpText={helpText}
+      errors={errors}
+      testId={testId}
+    >
+      <OTPInput
+        id={otpInputId}
+        value={value}
+        onChange={onChange}
+        numInputs={codeLength}
+        containerStyle={{ gap: "4px" }}
+        renderInput={(inputProps, index) => {
+          return (
+            <InternalInput
+              name={name}
+              inputProps={inputProps}
+              hasErrors={hasErrors}
+              disabled={disabled}
+              onFocus={onFocus}
+              onBlur={onBlur}
+              index={index}
+            />
+          );
+        }}
+      />
+    </Field>
+  );
+}
+
+OtpInput.DEFAULT_PROPS = DEFAULT_PROPS;
+
+OtpInput.propTypes = {
+  name: PropTypes.string.isRequired,
+  color: PropTypes.string,
+  codeLength: PropTypes.number,
+  isDisabled: PropTypes.bool,
+  shouldAutoFocus: PropTypes.bool,
+  value: PropTypes.string,
+  onChange: PropTypes.func,
+  validate: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
+  validateData: PropTypes.any,
+  label: PropTypes.string.isRequired,
+  helpText: PropTypes.node,
+  testId: PropTypes.string,
+};
+
+function InternalInput(props) {
+  const {
+    hasErrors,
+    onFocus,
+    onBlur,
+    index,
+    name,
+    disabled,
+    inputProps: { style, ...otpInputProps },
+  } = props;
+  const theme = useTheme();
+  const { inputColorMap } = useBackground();
+  const inputCSS = useResponsivePropsCSS(props, DEFAULT_PROPS, {
+    color: (propsAtBreakpoint, theme, bp) => {
+      const color = props.color ?? inputColorMap[bp];
+
+      return theme.otpInput.getCSS({
+        targetElement: "input",
+        color,
+      });
+    },
+  });
+
+  const internalOnFocus = useCallback(
+    (event) => {
+      otpInputProps.onFocus(event);
+      onFocus(event);
+    },
+    [otpInputProps, onFocus]
+  );
+
+  const internalOnBlur = useCallback(
+    (event) => {
+      otpInputProps.onBlur(event);
+      onBlur(event);
+    },
+    [otpInputProps, onBlur]
+  );
+
+  return (
+    <div
+      css={theme.input.getCSS({
+        targetElement: "inputContainer",
+      })}
+    >
+      <input
+        aria-invalid={hasErrors ? "true" : null}
```

Comment:
Is this a string true or boolean?

Reply:
Yea, this one is confusing. `aria-invalid` is string "true" when there is a validation error. 
However, when the data is valid we set `aria-invalid` to null, then in rendered html it will not provide key `aria-invalid` at all. This is an existing pattern from the other Basis components 😃 

![image](https://github.com/LatitudeFinancialOSS/basis/assets/4622799/6c72e7ae-25a1-4621-b30d-2ea0f95e280b)